### DISCUSSION TITLE
Files module failed after migrating from ublue-os/startingpoint

### DISCUSSION THREAD

**Discussion by miabbott** [blue-build/.github#14] [Support]
I have a custom image build that I migrated from the ublue-os/startingpoint template to the blue-build/template according to the instructions.  The commit where I made all the changes is here - https://github.com/miabbott/infidelicious-ublue/commit/7e9d855e98ca62e9449ade407bed8b0496f54846

But the build failed on the `files` module - https://github.com/miabbott/infidelicious-ublue/actions/runs/8124175950/job/22205373698

Since I didn't have anything under `config/files/usr`, I just used the nuclear option and removed the `files` module from my recipe.  This allowed my build to succeed - https://github.com/miabbott/infidelicious-ublue/actions/runs/8124214152/job/22205453199

I finally realized that in my repo, there was no `config/files/` directory present after I removed the `.just` files as part of the migration.

When I added an dummy `config/files/usr/.files` to my repo and re-enabled the `files` module, the build succeeded again - https://github.com/miabbott/infidelicious-ublue/actions/runs/8124318970/job/22205654715

---

I guess I am throwing up the warning flag that some users may run into this problem after the migration.  Perhaps the compiler could test if the `files` module is being used and test for the presence of `config/files/` before templating out the Containerfile?

**Comment by xynydev** [comment]
In my opinion, the correct thing for the file module is to fail if the file it is passed does not exist, and git does not keep empty directories by default. Thanks for documenting this issue for others who might encounter it! This could have been communicated more clearly in the migration guide.
